recipe/sitecustomize.py

```diff
@@ -0,0 +1,15 @@
+import site, sys, os
+
+dirs_to_add = []
+# Workaround for https://github.com/conda/conda/issues/14053
+if sys.abiflags != '':
+    dirs_to_add.append(os.path.join(sys.prefix, 'lib', f'python3.{sys.version_info[1]}', 'site-packages'))
+# Workaround for https://github.com/conda/conda/issues/10969
+dirs_to_add.append(os.path.join(sys.prefix, 'lib', f'python3.1', 'site-packages'))
+# A python version independent directory that ABI3 and noarch packages can use.
+# This is unused at the moment, but keeping it here for experimentation.
+dirs_to_add.append(os.path.join(sys.prefix, 'lib', f'python', 'site-packages'))
+
+for d in dirs_to_add:
+    if os.path.exists(d):
+        site.addsitedir(d)
```

Comment:
@jjhelmus, thoughts on this?

Reply:
I think these paths could also be added to sys.path using [`.pth` files](https://docs.python.org/3/library/site.html) which would leave `sitecustomize.py` available for other uses and would not break if `sitecustomize.py` was clobbered or deleted.

[Pex](https://pypi.org/project/pex/) loads modules from multiple directories by adding entries to sys.path. I've had experience with pex and most projects work fine with this setup. There are a few cases where the setup can fail where the packages work fine with a single site-packages directory:
* [Namespace packages](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/) that use different namespace conventions.  
* Packages that load data, config or other files from other packages by navigating the filesystem. For example loading a config file using `os.path.join(os.path.dirname(os.path.dirname(mod.__file__)), "mod_data", "config.yaml")`.  This type of access really should be using `importlib.resources`.

Personally, I see benefits and downsides to this approach. 

I can imagine there may be confusion to users and developers if they see multiple site-packages directories which contain files. Which one is the correct one? All of these directories need to be examined to determine which projects are available to Python. Using symlinks clarifies this some but leaves multiple directories.

On the other hand if conda/mamba/pixie/pip places files in the correct site-packages directory then the workaround directories do not exist with this appoarch. There is no confusion or uncertainty. With the symlink approach the symlinks are always there.

One other gotya is that conda-build will set $SP_DIR to the first directory matching `lib/python?.*` that is not a symlink, https://github.com/conda/conda-build/blob/24.7.1/conda_build/utils.py#L971-L982. With this approach a noarch: python package installed into a free-threading environment will result in the directory being set incorrectly to the the `python3.13` directory. The logic that conda-build uses to set `SP_DIR` can be made more robust to avoid this.